pkg/cri/server/image_pull.go

```diff
@@ -93,9 +94,13 @@ import (
 
 // PullImage pulls an image with authentication config.
 func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest) (*runtime.PullImageResponse, error) {
+	inProgresImagePulling.Inc()
+	defer inProgresImagePulling.Dec()
+	startTime := time.Now()
 	imageRef := r.GetImage().GetImage()
 	namedRef, err := distribution.ParseDockerRef(imageRef)
 	if err != nil {
+		imagePullingError.WithValues(imageRef).Inc()
```

Comment:
without looking too deeply into the code, can we do this in a defer like:

```
func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest) (_ *runtime.PullImageResponse, err error) {
...

defer {
  if err != nil {
    imagePullingError.WithValues(imageRef).Inc()
  }
}()
```

Reply:
Looks better.